Add package to {cpus,uncores} matching functions

sgruszka opened the pull request:
Allow to specify cpus and uncores based on package (socket) number.

yarda commented on tuned/profiles/functions/function_package2cpus.py:
```diff
@@ -0,0 +1,59 @@
+	"""
+
+	def __init__(self):
+		# 1 argument
```
Could it make sense to have 1 to infinite arguments? E.g. If I want to get devices for package 1 and 2:
```ini
devices=${f:package2cpus:1:2}
```
If such queries make sense, the constrain could be relaxed and the arguments processed in the loop.

yarda commented on tuned/profiles/functions/function_package2uncores.py:
```diff
@@ -0,0 +1,72 @@
+	"""
+
+	def __init__(self):
+		# 1 argument
```
Again, maybe relax maximal number of arguments? If it makes sense.

yarda commented on tuned/profiles/functions/function_package2uncores.py:
```diff
@@ -0,0 +1,72 @@
+			try:
+				package_id = int(value)
+			except ValueError:
+				continue
```
At least log.debug or maybe log.info/warn would be useful here, e.g. to detect case if kernel changes the interface.

yarda commented on tuned/profiles/functions/function_package2cpus.py:
```diff
@@ -0,0 +1,59 @@
+			try:
+				package_id = int(value)
+			except ValueError:
+				continue
```
I think at least log.debug could be useful here.

yarda commented:
Is it ready for merge, or still WIP (because it has WIP flag)? From my point of view it's ready. I added few comments that could improve it, but I can see no problem with this PR.

sgruszka commented:
> Is it ready for merge, or still WIP (because it has WIP flag)? From my point of view it's ready. I added few comments that could improve it, but I can see no problem with this PR.

Addressed the comments and remove WIP flag. Please review.

yarda commented on tuned/profiles/functions/function_package2cpus.py:
```diff
@@ -0,0 +1,63 @@
+				value = cmd.read_file(f)
+				package_id = int(value)
+			except ValueError:
+				log.warning("Failed to get package id for cpu '%s'" % cpu)
```
I think you could also log the content of the `value` which may help with debugging possible problems.

yarda commented on tuned/profiles/functions/function_package2uncores.py:
```diff
@@ -0,0 +1,75 @@
+			try:
+				package_id = int(value)
+			except ValueError:
+				log.warning("Failed to get package id for uncore '%s'" % uncore)
```
I think you could also log the content of the `value` which may help with debugging possible problems.

yarda reviewed (commented):
It's not blocker, but I think it's worth fixing.

yarda commented:
CI failure seems unrelated to this PR.

sgruszka commented:
Addressed the comments, please review.

yarda reviewed (approved):
LGTM, thanks.

CI failures are unrelated:
- Centos 7 CI failure is caused by dropped EPEL-7, it's being worked on.
- Fedora rawhide failure is caused by systemd rate limiting, workaround will be enabled in beakerlib/tuned library.
